harmonica/filters/_filters.py

```diff
@@ -62,3 +62,598 @@ def derivative_upward_kernel(fft_grid, order=1):
+        magnetization_declination = declination
+    # Catch the dims of the Fourier transformed grid
+    dims = fft_grid.dims
+    # Grab the coordinates of the Fourier transformed grid
+    freq_easting = fft_grid.coords[dims[1]]
+    freq_northing = fft_grid.coords[dims[0]]
+    # Convert frequencies to wavenumbers
+    k_easting = 2 * np.pi * freq_easting
+    k_northing = 2 * np.pi * freq_northing
+    # Compute the filter for reduction to pole in frequency domain
+    da_filter = _get_rtp_filter(
+        k_easting,
+        k_northing,
+        inclination,
+        declination,
+        magnetization_inclination,
+        magnetization_declination,
+    )
+    # Deal with inf and nan value
+    da_filter.data = np.nan_to_num(da_filter.data, posinf=0, nan=0)
+    return da_filter
+
+
+def pseudo_gravity_kernel(
+    fft_grid,
+    inclination=90,
+    declination=0,
+    magnetization_inclination=None,
+    magnetization_declination=None,
+    f=50000,
+):
+    r"""
+    Filter for pseudo gravity in frequency domain
+
+    Return a :class:`xarray.DataArray` with the values of the frequency domain
+    filter for computing the pseudo gravity. The filter is built upon the
+    frequency coordinates of the passed ``fft_grid`` and is defined as follows:
+
+    .. math::
+
+            g(\mathbf{k}) = \frac{149.8}{f|\mathbf{k}|}\times
+            {reduction\_to\_pole\_kernel}
+
+
+    where :math:`\mathbf{k}` is the wavenumber vector
+    (:math:`\mathbf{k} = 2\pi \mathbf{f}`
+
+    Parameters
+    ----------
+    fft_grid : :class:`xarray.DataArray`
+        Total Magnetic Intensity or Reduction To Pole magnetic field.
+        Please set inclination = 90 for Reduction To Pole magnetic field.
+        Array with the Fourier transform of the original grid.
+        Its dimensions should be in the following order:
+        *freq_northing*, *freq_easting*.
+        Use :func:`xrft.xrft.fft` and :func:`xrft.xrft.ifft` functions to
+        compute the Fourier Transform and its inverse, respectively.
+    inclination : float in degrees
+        The inclination inducing Geomagnetic field. Default is 90 degree for
+        RTP field.
+    declination : float in degrees
+        The declination inducing Geomagnetic field. Default is 0 degree for
+        RTP field.
+    magnetization_inclination : float in degrees
+        The inclination of the total magnetization of the anomaly source.
+        Default is i, neglecting remanent magnetization and
+        self demagnetization.
+    magnetization_declination : float in degrees
+        The declination of the total magnetization of the anomaly source.
+        Default is d, neglecting remanent magnetization and
+        self demagnetization.
+    f : float or 2d-array
+        Ambient field in the study area. It can use the mean ambinent field
+        value in the study area or the real ambient field value in all
+        locations. Default is 50,000 nT.
```

Comment:
I don't quite like the `f` name for this parameter. Could we rename it?

Reply:
Sure.. Renamed to `ambient field`